### ISSUE TITLE
Current setup vs full repository project templates?

### ISSUE THREAD

**Issue by ejgallego** [coq-community/templates#105]
Hi folks,

this is a follow up from the discussion at https://coq.zulipchat.com/#narrow/stream/240550-Dune-devs.20.26.20users/topic/.E2.9C.94.20Dune.2C.20_CoqProject.20and.20tests ; indeed, it seems to me that the current "template" setup may be a bit limited (see also #95 ) , in particular as to setup full projects such as the plugin template.

Moreover, we have the current alternative Coq project templates in the wild:

- https://github.com/ejgallego/coq-plugin-template
- https://github.com/palmskog/coq-program-verification-template

How should we organize things? Should we move all the templates to `coq-template-foo` and have this repository just be an index to all the existing templates, as suggested in the thread?

**Comment by Zimmi48** [comment]
IMHO you are comparing two very different things. On the one hand, this repository should be considered as a tool to generate *and maintain* some standard files (e.g. CI) for both new *and preexisting* projects. On the other hand, the repositories you point to are example projects that demonstrate how to do certain things and can be used as a template to create a new project. (BTW, given that GitHub supports marking a repository as a "template", maybe this should be done for these two repos.) If we wanted, the latter could even include a `meta.yml` file and include some files that would be auto-generated with the templates here.

**Comment by palmskog** [comment]
I fully agree with the analysis of Théo, and I even have a local `meta.yml` for program verification template (which I didn't want to distract people with).

But more generally, how do you feel about hosting template repos like Emilio's and mine in coq-community @Zimmi48? Maybe the proper place for this discussion is in the manifesto repo?

**Comment by Zimmi48** [comment]
> But more generally, how do you feel about hosting template repos like Emilio's and mine in coq-community @Zimmi48? Maybe the proper place for this discussion is in the manifesto repo?

It would totally make sense to host these repos in coq-community indeed! And sure, that would probably deserve a discussion in the manifesto repo. Should we just transfer the current issue over there (since GitHub supports that)? Or is it worth keeping as a separate issue?
